wimm.Secundatives/Extensions/MaybeExtensions.cs

```diff
@@ -0,0 +1,79 @@
+    /// </summary>
+    public static class MaybeExtensions
+    {
+
+        /// <summary>
+        /// Extension to indicate that a value missing from a <see cref="Maybe{T}"/> is unrecoverable. 
+        /// Unwraps the <see cref="Maybe{T}"/> if it exists or throws a <see cref="ExpectException"/>
+        /// </summary>
+        /// <typeparam name="T"> The type of value contained with the <paramref name="maybe"/> </typeparam>
+        /// <param name="maybe"> The <see cref="Maybe{T}"/> to be inspected </param>
+        /// <exception cref="ExpectException"> <paramref name="maybe"/> contains no value </exception>
+        /// <returns> The <typeparamref name="T"/> held within <paramref name="maybe"/> if it exists. </returns>
+        public static T Expect<T>(this Maybe<T> maybe)
+        {
+            return maybe.Expect($"Expected value in {nameof(Maybe<T>)}");
+        }
+
+
+        /// Extension to indicate that a value missing from a <see cref="Maybe{T}"/> is unrecoverable. 
+        /// Unwraps the <see cref="Maybe{T}"/> if it exists or throws a <see cref="ExpectException"/>
+        /// with a custom message.
+        /// <typeparam name="T"> The type of value contained with the <paramref name="maybe"/> </typeparam>
+        /// <param name="maybe"> The <see cref="Maybe{T}"/> to be inspected </param>
+        /// <param name="message"> The message to be contained in the thrown <see cref="ExpectException"/> 
+        /// if <paramref name="maybe"/> contains no value </param>
+        /// <exception cref="ExpectException"> <paramref name="maybe"/> contains no value </exception>
+        /// <returns> The <typeparamref name="T"/> held within <paramref name="maybe"/> if it exists. </returns>
+        public static T Expect<T>(this Maybe<T> maybe, string message)
+        {
+            if (message == null)
+                throw new ArgumentNullException(nameof(message));
+            if (string.IsNullOrWhiteSpace(message))
+                throw new ArgumentException("Parameter must not be whitespace or empty", nameof(message));
+
+            return maybe.Exists ? maybe.Value : throw new ExpectException(message);
+        }
+
+
+        /// <summary>
+        /// Unwraps the <see cref="Maybe{T}"/> if possible and returns a provided default otherwise
+        /// </summary>
+        /// <typeparam name="T"> The type of value contained with the <paramref name="maybe"/> </typeparam>
+        /// <param name="maybe"> The <see cref="Maybe{T}"/> to be unwrapped </param>
+        /// <param name="defaultValue"> An instance of <typeparamref name="T"/> to be returned if unwrapping is not 
+        /// possible </param>
+        /// <returns> The <typeparamref name="T"/> held within <paramref name="maybe"/> if it exists. Otherwise
+        /// <paramref name="defaultValue"/> </returns>
+        public static T UnwrapOr<T>(this Maybe<T> maybe, T defaultValue)
+        {
+            return maybe.Exists ? maybe.Value : defaultValue;
```

Comment:
Did you consider disallowing `null` for `defaultValue`? I'm fine with either but curious how you decided.

Reply:
I considered it but since there's no guarantee that all methods that you call understand what maybes are you might interact with other APIs that need nulls for non-existent values or to signify something or you might be working within an interface that doesn't use Maybe<T> and is supposed to return null for non-existence. I didn't think there was any real benefit to limiting it given that